mdsuite/transformations/map_molecules.py

```diff
@@ -119,286 +118,178 @@ def _run_dependency_check(self):
-            self.reference_molecules[item] = {}
-            mol, species = MolecularGraph(
-                self.experiment,
-                from_smiles=True,
-                smiles_string=self.molecules[item]["smiles"],
-            ).build_smiles_graph()
-            self.reference_molecules[item]["species"] = list(species)
-            self.reference_molecules[item]["graph"] = mol
-            self.reference_molecules[item]["mass"] = self._get_molecular_mass(species)
-
-    def _get_molecular_mass(self, species_dict: dict) -> float:
-        """
-        Get the mass of a SMILES molecule based on experiment data.
+        Build the reduced mass dictionary.
+
+        This is a dictionary of reduced masses for each species in a molecule,
+        i.e m_species / m_molecule. These are used in the COM positions computation
+        later.
 
         Parameters
         ----------
-        species_dict : dict
-                Dictionary of species information along with number of species
-                present in the system.
+        species : list
+                List of species to include in the dict.
+
         Returns
         -------
-        mass : float
-                mass of the molecule
+        reduced_mass_dict : dict
+                Dictionary of reduced masses for each species.
         """
-        mass = 0.0
-        for item in species_dict:
-            mass += self.experiment.species[item]["mass"][0] * species_dict[item]
+        reduced_mass_dict = {}
+        for item in species:
+            reduced_mass_dict[item] = (
+                self.experiment.species[item]["mass"][0] / molecular_mass
+            )
 
-        return mass
+        return reduced_mass_dict
 
-    def _build_configuration_graphs(self):
+    def _map_molecules(self, molecular_graph: MolecularGraph):
         """
-        Build the adjacency graphs for the configurations
+        Map the molecules and save the data in the database.
 
         Returns
         -------
-        Nothing.
+        Updates the database.
         """
-        for item in self.reference_molecules:
-            self.adjacency_graphs[item] = {}
-            mol = MolecularGraph(
-                self.experiment,
-                from_configuration=True,
-                species=self.reference_molecules[item]["species"],
-            )
-            self.adjacency_graphs[item]["graph"] = mol.build_configuration_graph(
-                cutoff=self.molecules[item]["cutoff"]
-            )
+        molecule_name = molecular_graph.molecule_name
+        molecules = self.experiment.molecules
+        molecules[molecule_name] = {}
+        molecules[molecule_name]["n_particles"] = molecular_graph.n_molecules
 
-    def _get_molecule_indices(self):
-        """
-        Collect the indices of the molecules.
+        molecules[molecule_name]["mass"] = molecular_graph.molecular_mass
+        molecules[molecule_name]["groups"] = molecular_graph.molecular_groups
```

Comment:
Dataclass would fit nicely

Reply:
Yes so this is again related to what goes into the experiment molecules property, it is the same thing here as well.